meta-sokol-flex-staging: libcoap: Introduce libcoap recipe

muhamzai opened the pull request:
Introduce libcoap recipe which belongs to meta-oe in
meta-networking so we can drop our maintained version
of libcoap recipe.

JIRA-ID: SB-20025

Signed-off-by: Muhammad Hamza <muhammad_hamza@mentor.com>

abelal reviewed (changes requested):
Rather than "Introduce libcoap recipe" we should say "backport libcoap from meta-networking". In the commit message please share the exact link from where this is being copied over.

muhamzai commented:
> Rather than "Introduce libcoap recipe" we should say "backport libcoap from meta-networking". In the commit message please share the exact link from where this is being copied over.

Fixed and updated.
